src/main/java/com/kosherjava/zmanim/hebrewcalendar/TefilaRules.java

```diff
@@ -512,6 +512,49 @@ public boolean isMizmorLesodaRecited(JewishCalendar jewishCalendar) {
 	    return true;
 	}
 	
+	/**
+	 * Returns if <em>Selichos</em> is recited on the day in question. 
+	 * @param jewishCalendar  the Jewish calendar day.
+	 * @param isSefardi  whether this is the Sefardi or Ashkenazi minhag.
+	 * @return if <em>Selichos</em> is recited.
+	 * 
+	 */
+	public boolean isSelichosRecited(JewishCalendar jewishCalendar, boolean isSefardi) {
+		int holidayIndex = getYomTovIndex();
+
+		// All fast days except Tisha B'Av say Selichos even if they fall out on Shabbos (only Yom Kippur can)
+		if (holidayIndex == JewishCalendar.SEVENTEEN_OF_TAMMUZ || holidayIndex == JewishCalendar.YOM_KIPPUR
+				|| holidayIndex == JewishCalendar.FAST_OF_GEDALYAH || holidayIndex == JewishCalendar.TENTH_OF_TEVES || holidayIndex == JewishCalendar.FAST_OF_ESTHER) {
+			return true;
+		}
+
+		// We don't say Selichos on Shabbos
+		if (jewishCalendar.getDayOfWeek() == Calendar.SATURDAY) {
+			return false;

```

Comment:
At least for sepharadim, even if Yom Kippur falls on Shabbat, we still have selichot.